peripherals/aes/aes_wl.yaml

```diff
@@ -0,0 +1,69 @@
+AES:
+  CR:
+    NPBLB: [0, 15]
+    KEYSIZE:
+      b128: [0, "128 bits"]
+      b256: [1, "256 bits"]
+    GCMPH:
+      Init: [0, "Init phase"]
+      Header: [1, "Header phase"]
+      Payload: [2, "Payload phase"]
+      Final: [3, "Final phase"]
+    DMAOUTEN:
+      Disabled: [0, "Disable DMA Output"]
+      Enabled: [1, "Enabled DMA Output"]
+    DMAINEN:
+      Disabled: [0, "Disable DMA Input"]
+      Enabled: [1, "Enable DMA Input"]
+    ERRIE:
+      Disabled: [0, "Disable (mask) error interrupt"]
+      Enabled: [1, "Enable error interrupt"]
+    CCFIE:
+      Disabled: [0, "Disable (mask) CCF interrupt"]
+      Enabled: [1, "Enable CCF interrupt"]
+    ERRC:
+      _write:
+          Clear: [1, "Clear RDERR and WRERR flags"]
+    CCFC:
+      _write:
+        Clear: [1, "Clear computation complete flag"]
+    DATATYPE:
+      None: [0, "Word"]
+      HalfWord: [1, "Half-word (16-bit)"]
+      Byte: [2, "Byte (8-bit)"]
+      Bit: [3, "Bit"]
+    EN:
+      Disabled: [0, "Disable AES"]
+      Enabled: [1, "Enable AES"]
+    CHMOD:
+      ECB: [0, "Electronic codebook (ECB)"]
+      CBC: [1, "Cipher-block chaining (CBC)"]
+      CTR: [2, "Counter mode (CTR)"]
+      GCM: [3, "Galois counter mode (GCM) and Galois message authentication code (GMAC) 100: Counter with CBC-MAC (CCM)"]
+    MODE:
+      Mode1: [0, "Mode 1: encryption"]
+      Mode2: [1, "Mode 2: key derivation (or key preparation for ECB/CBC decryption)"]
+      Mode3: [2, "Mode 3: decryption"]
```

Comment:
There is an undocumented mode 4 here.

The reference manual looks like this:
![image](https://user-images.githubusercontent.com/7845120/120935343-2e1e8900-c6b7-11eb-877c-484949beaa34.png)

but in code generated from Cube-MX there is a fourth mode:

```c
#define CRYP_OPERATINGMODE_ENCRYPT                   0x00000000U     /*!< Encryption mode(Mode 1)  */
#define CRYP_OPERATINGMODE_KEYDERIVATION             AES_CR_MODE_0   /*!< Key derivation mode  only used when performing ECB and CBC decryptions (Mode 2) */
#define CRYP_OPERATINGMODE_DECRYPT                   AES_CR_MODE_1   /*!< Decryption    (Mode 3)    */
#define CRYP_OPERATINGMODE_KEYDERIVATION_DECRYPT     AES_CR_MODE     /*!< Key derivation and decryption only used when performing ECB and CBC decryptions (Mode 4) */
```

The fourth mode is rather essential for fast ECB decryption, I have tested this undocumented mode and it does work:
code: https://github.com/newAM/stm32wl-hal/blob/main/aes/src/lib.rs#L395
tests: https://github.com/newAM/stm32wl-hal/blob/main/aes-testsuite/tests/test.rs#L83

Reply:
Thanks for the catch and the testing. I will add it with an Undocumented tag.